common/s3util.cpp

```diff
@@ -41,11 +50,77 @@ using Aws::S3::Model::ListObjectsRequest;
 using Aws::S3::Model::HeadObjectRequest;
 using Aws::Utils::RateLimits::DefaultRateLimiter;
 
+
 namespace common {
 
+//TODO(angxu) currently page size is used as buffer size,
+// we might want to use a larger buffer size if write is
+// too slow
+const uint32_t kPageSize = getpagesize();
+
+DirectIOFile::DirectIOFile(const string& file_path)
+    : fd_(-1)
+    , file_size_(0)
+    , buffer_()
+    , offset_(0) {
+  if (posix_memalign(&buffer_, kPageSize, kPageSize) != 0) {
+    LOG(ERROR) << "Failed to allocate memaligned buffer, errno = " << errno;
+    return;
+  }
+
+  int flag = O_WRONLY | O_TRUNC | O_CREAT | O_DIRECT;
+  fd_ = open(file_path.c_str(), flag , 640);
```

Comment:
I got these from man 2 open. do they work?

The following symbolic constants are provided for mode:

              S_IRWXU  00700 user (file owner) has read, write and execute permission

              S_IRUSR  00400 user has read permission

              S_IWUSR  00200 user has write permission

              S_IXUSR  00100 user has execute permission

              S_IRWXG  00070 group has read, write and execute permission

              S_IRGRP  00040 group has read permission

              S_IWGRP  00020 group has write permission

              S_IXGRP  00010 group has execute permission

              S_IRWXO  00007 others have read, write and execute permission

              S_IROTH  00004 others have read permission

              S_IWOTH  00002 others have write permission

              S_IXOTH  00001 others have execute permission